test/unit/overlay.js

```diff
@@ -0,0 +1,62 @@
+'use strict';
+
+var fixtures = require('../fixtures');
+var fs = require('fs');
+var sharp = require('../../index');
+
+sharp.cache(0);
+
+// Main
+describe('Overlays', function() {
+  it('Overlay transparent PNG on solid background', function(done) {
+    sharp(fixtures.inputPngOverlayLayer0)
+      .overlayWith(fixtures.inputPngOverlayLayer1)
+      .toBuffer(function (error, data, info) {
+        if (error) return done(error);
```

Comment:
Minor: I noticed you use `if (error) throw error;` in your unit tests but doesn’t mixing synchronous `throw` inside an async callback cause issues? I am fine doing whatever you prefer in terms of style. Just let me know.